src/DotNetEnv/Parser.cs

```diff
@@ -15,10 +15,26 @@ private static bool IsComment(string line)
             return line.Trim().StartsWith("#");
         }
 
+        private static string GetQuotedValue(string input, string delimeter)
+        {
+            if (input.Contains(delimeter))
+            {
+                int quoteStart = input.IndexOf(delimeter);
+                int quoteEnd = input.LastIndexOf(delimeter)+1;
+                return input.Substring(quoteStart, quoteEnd);
+            }
+            return input;
+        }
         private static string RemoveInlineComment(string line)
         {
-            int pos = line.IndexOf('#');
-            return pos >= 0 ? line.Substring(0, pos) : line;
+            string value = GetQuotedValue(line, "\"");
+            value = GetQuotedValue(value, "\'");
+
+            if (IsQuoted(value))
```

Comment:
Your tests look good so I'd like a walk through because I remain concerned that this will break something that I do not have time right now to think of: you have removed the quotes and *then* check if it is quoted, which will always be false, it seems to me.  Again, your tests are what I want to see so it must work, but I would like you to expalin to me how so I can feel more confident about this logic.  Thanks